ui/src/views/Hooks/ListHooks/index.jsx

```diff
@@ -46,25 +48,80 @@ export default class ListHooks extends Component {
     });
   };
 
+  updateExpanded() {
+    this.query = qs.parse(this.props.location.search.slice(1));
+    const {
+      data: { hookGroups },
+    } = this.props;
+    const hookSearch = this.query.search;
+    const expandedHookGroups = hookGroups
+      ? hookGroups
+          .filter(
+            group =>
+              group.hooks.filter(hook => hook.hookId.includes(hookSearch))
+                .length !== 0
+          )
+          .map(group => group.hookGroupId)
+      : [];
+
+    this.setState({ expanded: expandedHookGroups, selected: [] });
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: [], selected: [] };
+    this.updateExpanded();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.location.search.slice(1) !== this.props.location.search.slice(1)
+    ) {
+      this.updateExpanded();
+    }
+  }
+
   render() {
     const {
       classes,
       description,
       data: { loading, error, hookGroups },
     } = this.props;
-    const query = qs.parse(this.props.location.search.slice(1));
-    const hookSearch = query.search;
+    const hookSearch = this.query.search;
     const tree = hookGroups
-      ? hookGroups.map(group => ({
-          value: group.hookGroupId,
-          nodes: group.hooks.map(hook => ({
-            value: hook.hookId,
-            href: `/hooks/${group.hookGroupId}/${encodeURIComponent(
-              hook.hookId
-            )}`,
-          })),
-        }))
+      ? hookGroups.map(group => {
+          const nodes = group.hooks.map(hook => (
+            <TreeItem
+              key={hook.hookId}
+              nodeId={hook.hookId}
+              label={
+                <Link
+                  to={`/hooks/${group.hookGroupId}/${encodeURIComponent(
+                    hook.hookId
+                  )}`}>
+                  {hook.hookId}
+                </Link>
+              }
+            />
+          ));
+
+          return (
+            <TreeItem
+              key={group.hookGroupId}
+              nodeId={group.hookGroupId}
+              label={group.hookGroupId}>
+              {nodes}
+            </TreeItem>
+          );
+        })
       : [];
+    const handleToggle = (event, nodeIds) => {
+      this.setState({ expanded: nodeIds });
+    };
+
+    const handleSelect = (event, nodeIds) => {
+      this.setState({ selected: nodeIds });
+    };
```

Comment:
These two functions (handleToggle and handleSelect) should be with the other class methods rather than defined in render. Alternatively, you can define them inline on lines 141 and 142 if they're only used once: `expanded={(event, nodeIds) => this.setState({ selected: nodeIds })`.

Reply:
Another option, reduce the duplicate code by combining these two functions into one: `handleUpdates = (stateValue, nodeIds) => this.setState({ [stateValue]: nodeIds })` and then `onNodeSelect={(nodeIds) => this.handleUpdates('selected', nodeIds)}`. You basically use a string to point to the correct key on the state object.

Reply:
did something similar, let me know if it is good enough